ingenialink/pdo.py

```diff
@@ -306,107 +321,80 @@ def reset_tpdo_mapping(self) -> None:
+    def map_rpdos(self) -> None:
+        """Map the RPDO registers into the servo drive.
+        It takes the first available RPDO assignment slot of the drive.
 
         Raises:
             ILError: If there are no available PDOs.
         """
-        if len(self._rpdo_maps) + 1 > self.AVAILABLE_PDOS:
-            raise ILError("Could not add a new RPDO map, there are no available PDOs.")
-        self._rpdo_maps.append(rpdo_map)
-        map_index = len(self._rpdo_maps) - 1
-
+        if len(self._rpdo_maps) > self.AVAILABLE_PDOS:
+            raise ILError(
+                f"Could not map the RPDO maps, received {len(self._rpdo_maps)} PDOs and only"
+                " {self.AVAILABLE_PDOS} are available"
+            )
         self.write(self.RPDO_ASSIGN_REGISTER_SUB_IDX_0, len(self._rpdo_maps))
-        self.write(self.RPDO_MAP_REGISTER_SUB_IDX_0[map_index], len(rpdo_map.items))
-        self.write(
-            self.RPDO_MAP_REGISTER_SUB_IDX_1[map_index],
-            rpdo_map.items_mapping.decode("utf-8"),
-            complete_access=True,
-        )
+        custom_map_index = 0
+        rpdo_assigns = b""
+        for rpdo_map in self._rpdo_maps:
+            if rpdo_map.map_register_index is not None:
+                rpdo_assigns += rpdo_map.map_register_index_bytes
+            self.write(self.RPDO_MAP_REGISTER_SUB_IDX_0[custom_map_index], len(rpdo_map.items))
+            self.write(
+                self.RPDO_MAP_REGISTER_SUB_IDX_1[custom_map_index],
+                rpdo_map.items_mapping.decode("utf-8"),
+                complete_access=True,
+            )
+            rpdo_map.map_register_index = self.RPDO_MAP_REGISTER_SUB_IDX_0[custom_map_index].idx
+            rpdo_assigns += rpdo_map.map_register_index_bytes
+            custom_map_index += 1
         self.write(
-            self.RPDO_ASSIGN_REGISTER_SUB_IDX_1[map_index],
-            self.RPDO_MAP_REGISTER_SUB_IDX_0[map_index].idx,
+            self.RPDO_ASSIGN_REGISTER_SUB_IDX_1,
+            rpdo_assigns,
             complete_access=True,
         )
-        rpdo_map.map_register_index = self.RPDO_MAP_REGISTER_SUB_IDX_0[map_index].idx
-
-    def add_tpdo_map(self, tpdo_map: TPDOMap) -> None:
-        """Add a new TPDO map into the drive.
 
-        It takes the first available (not mapped yet) TPDO assignment slot of the drive.
-
-        Args:
-            tpdo_map: TPDO map to be added.
+    def map_tpdos(self) -> None:
+        """Map the TPDO registers into the servo drive.
+        It takes the first available TPDO assignment slot of the drive.
 
         Raises:
             ILError: If there are no available PDOs.
         """
-        if len(self._tpdo_maps) + 1 > self.AVAILABLE_PDOS:
-            raise ILError("Could not add a new TPDO map, there are no available PDOs.")
-        self._tpdo_maps.append(tpdo_map)
-        map_index = len(self._tpdo_maps) - 1
-
+        if len(self._tpdo_maps) > self.AVAILABLE_PDOS:
```

Comment:
Why are the `_tpdo_maps` and `_rpdo_maps` lists, when we can only have a single PDO map?

Reply:
in the future we will have support for multiple pdo maps like many ethercat devices

Reply:
It will change for now FW, that should be defined in dicctionaty